Sources/MongoSwift/BSON/BSONEncoder.swift

```diff
@@ -803,10 +828,16 @@ private class MutableDictionary: BSONValue {
     }
 
     /// Converts self to a `BSONDocument` with equivalent key-value pairs.
-    fileprivate func toDocument() -> BSONDocument {
+    fileprivate func toDocument() throws -> BSONDocument {
         var doc = BSONDocument()
         for i in 0..<self.keys.count {
-            doc[self.keys[i]] = self.values[i].bson
+            try convertingBSONErrors {
```

Comment:
do we need this `convertingBSONErrors`, given that at the top level we are converting everything to an `EncodingError` anyway?